tekton/pipeline_run.go

```diff
@@ -112,7 +113,11 @@ func (r *ReleasePipelineRun) WithApplicationSnapshot(snapshot *v1alpha1.Applicat
 	// add something like a `Complete` function that returns the final object and error.
 	snapshotString, _ := json.Marshal(snapshot.Spec)
 
-	r.WithExtraParam(snapshot.Kind, tektonv1beta1.ArrayOrString{
+	// Get snapshot.Kind runes to make the first letter lowercase
+	snapshotRunes := []rune(snapshot.Kind)
+	snapshotRunes[0] = unicode.ToLower(snapshotRunes[0])
```

Comment:
Just an FYI.
There used to be a function for that, but it was depreciated.
https://pkg.go.dev/strings#Title

Reply:
(resolve conversation once you read, this is just informational)

Reply:
Thanks. I saw another library for doing the same, but as it is an small change I went for this alternative.